File: problems/lengthOfLastWord/python/length_of_last_word.py
class Solution:
    """
    Complexity
    ----------
    - Time complexity: O(n)
    - Space complexity: O(1)
    """

    def lengthOfLastWord(self, s: str) -> int:
        # Split the string by spaces and get the last word,
        # then return its length
        return len(s.split()[-1])


if __name__ == "__main__":
    s = "Hello World"

    print(f"Solution: {Solution().lengthOfLastWord(s)}")

File: problems/lengthOfLastWord/c++/length_of_last_word.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        int length = 0;
        int i = s.length() - 1;

        // Skip trailing spaces
        while (i >= 0 && s[i] == ' ') {
            i--;
        }

        // Count the length of the last word
        while (i >= 0 && s[i] != ' ') {
            length++;
            i--;
        }

        return length;
    }
};

int main() {
    string word = "Hello World";

    Solution obj;
    cout << "The length of the last word is: " << obj.lengthOfLastWord(word) << endl;

    return 0;
}
